diff --git a/client/modules/Dashboard/UserDashboard/components/TopicContent.js b/client/modules/Dashboard/UserDashboard/components/TopicContent.js
index cf7aaca..2f8c9e8 100755
--- a/client/modules/Dashboard/UserDashboard/components/TopicContent.js
+++ b/client/modules/Dashboard/UserDashboard/components/TopicContent.js
@@ -180,22 +180,22 @@ class TopicContent extends Component{
 			                </div>
 			              </Link>
 			            </li>
-						    }	else if(fileType == 'video'){
-						    	let src = "/uploads/"+data.fileName;
-						    	return <li key={data._id}>
-			              <Link>
-			                <div className={styles.videoThubBox}>
-			                	<video src ={src} width="100" height="100" style={video} key={fileName} onClick={this.handleImage.bind(this,{fileName},{fileType})}/>
-			                </div>
-			              </Link>
-			            </li>
-						    }else if(fileType == 'link'){
+						    // }	else if(fileType == 'video'){
+						    // 	let src = "/uploads/"+data.fileName;
+						    // 	return <li key={data._id}>
+			       //        <Link>
+			       //          <div className={styles.videoThubBox}>
+			       //          	<video src ={src} width="100" height="100" style={video} key={fileName} onClick={this.handleImage.bind(this,{fileName},{fileType})}/>
+			       //          </div>
+			       //        </Link>
+			       //      </li>
+						    }else if(fileType == 'video'){
 						    	let src = "http://img.youtube.com/vi/"+data.fileName+"/1.jpg";
 						      fileName = "https://www.youtube.com/watch?v="+data.fileName;
 						    	return <li key={data._id}>
 			              <Link>
 			                <div className={styles.videoThubBox}>
-			                	<img src={src} width="100" height="100" key={fileName}/>
+			                	<img src={src} width="100" height="100" key={fileName} onClick={this.handleImage.bind(this,{fileName},{fileType})}/>
 			                </div>
 			              </Link>
 			            </li>
@@ -321,9 +321,9 @@ class TopicContent extends Component{
 						<Title>{this.state.fileName}</Title>
 					</Header>
 					<Body>
-						{this.state.fileType == "video"					
-						?<video src={fileURL} width="300" height="300" controls/> 
-							:<img src={fileURL} width="300" height="300"/>
+						{this.state.fileType == "video"				
+						?<video src={this.state.fileName} width="565" height="300" controls/> 
+							:<img src={fileURL} width="565" height="300"/>
 						}
 					</Body>
 					<Footer>
