// fs-io/index.js
'use strict';

var base = new (require('base-io'));

base
.import(function(){
	var self = this;

	var methods = {
		appendFile: 3,
		open: 3,
		close: 1,
		ftruncate: 2,
		chown: 3,
		fchown: 3,
		lchown: 3,
		chmod: 2,
		fchmod: 2,
		lchmod: 2,
		stat: 1,
		fstat: 1,
		lstat: 1,
		link: 2,
		symlink: 3,
		readlink: 1,
		realpath: 2,
		unlink: 1,
		rmdir: 1,
		mkdir: 2,
		readdir: 1,
		utimes: 3,
		futimes: 3,
		fsync: 1,
		write: 5,
		read: 5,
		writeFile: 3,
		readFile: 2
	};

	Object.keys(methods).forEach((name) => {
		self[name] = function(){
			var args = Array.prototype.slice.call(
				arguments,
				0,
				methods[name]
			);

			if (args[0] && typeof args[0] === 'string') {
				args[0] = this.normalize(args[0]);
			}

			return new self.promise((resolve,reject) => {
				self.fs[name].apply(
					self.fs,
					args.concat((error,result) => {
						if (error) {
							reject(error);
						}

						resolve(result);
					})
				);
			});
		};
	});
})
.root(__dirname)
.scan('async');

module.exports = new (base.export());
